### PR TITLE
[PAY-1592] Wire up USDC purchase flow on mobile

### PR DESCRIPTION
**Pull request by dharit-tan**
### Description
E2e usdc purchase on mobile!

Depends on https://github.com/AudiusProject/audius-client/pull/3873/files

### Dragons

- Existing balance not handled yet
- Might be some more error scenarios that I missed

### How Has This Been Tested?

https://github.com/AudiusProject/audius-client/assets/3893871/86e2fac3-43cd-4050-8177-7fe1a9f3ec96

https://github.com/AudiusProject/audius-client/assets/3893871/c6ea01e4-b70e-48d0-ad09-a8eb153d5b2f

### How will this change be monitored?

*For features that are critical or could fail silently please describe the monitoring/alerting being added.*

### Feature Flags ###

*Are all new features properly feature flagged? Describe added feature flags.*

### CODE REVIEW

**Comment by audius-infra** [comment]
It looks like there may be some changes to native mobile code, which requires triggering a full app release. Please follow the instructions here: https://www.notion.so/audiusproject/When-to-bump-app-version-2644a8f772364a4d91f44abcba44ce0b?pvs=4. cc @nicoback2 @sliptype

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/rt-wire-up-usdc-mobile

**Comment by audius-infra** [comment]
It looks like there may be some changes to native mobile code, which requires triggering a full app release. Please follow the instructions here: https://www.notion.so/audiusproject/When-to-bump-app-version-2644a8f772364a4d91f44abcba44ce0b?pvs=4. cc @nicoback2 @sliptype

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/rt-wire-up-usdc-mobile

**Comment by audius-infra** [comment]
It looks like there may be some changes to native mobile code, which requires triggering a full app release. Please follow the instructions here: https://www.notion.so/audiusproject/When-to-bump-app-version-2644a8f772364a4d91f44abcba44ce0b?pvs=4. cc @nicoback2 @sliptype

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/rt-wire-up-usdc-mobile

File: packages/common/src/store/index.ts [hunk 1]
```diff
@@ -35,3 +35,4 @@ export * from './solana'
 export * from './playlist-updates'
 export * from './saved-collections'
 export * from './confirmer'
+export * from './purchase-content'
```
Comment (schottra) [inline packages/common/src/store/index.ts:38]: I expect you will have a conflict :-)

File: packages/common/src/store/ui/stripe-modal/sagas.ts [hunk 2]
```diff
@@ -34,13 +34,13 @@ function* handleStripeSessionChanged({
...
     }
-    yield* put(setVisibility({ modal: 'StripeOnRamp', visible: false }))
+    yield* put(setVisibility({ modal: 'StripeOnRamp', visible: 'closing' }))
   }
 }
 
 function* handleCancelStripeOnramp() {
   const { onrampCanceled } = yield* select(getStripeModalState)
-  yield* put(setVisibility({ modal: 'StripeOnRamp', visible: false }))
+  yield* put(setVisibility({ modal: 'StripeOnRamp', visible: 'closing' }))
```
Comment (schottra) [inline packages/common/src/store/ui/stripe-modal/sagas.ts:43]: Is this from the recent modal changes @rickyrombo did? Or what motivated changing this to a string?

File: packages/mobile/src/components/premium-track-purchase-drawer/PremiumTrackPurchaseDrawer.tsx [hunk 3]
```diff
@@ -58,94 +83,102 @@ const useStyles = makeStyles(({ spacing, typography, palette }) => ({
...
+              <View style={styles.payToUnlockTitleContainer}>
+                <Text weight='heavy' textTransform='uppercase' fontSize='small'>
+                  {messages.payToUnlock}
+                </Text>
+                <LockedStatusBadge locked />
+              </View>
+              <Text>
+                {messages.disclaimer(
+                  <Text
+                    onPress={() => console.log('terms of use pressed')}
```
Comment (schottra) [inline packages/mobile/src/components/premium-track-purchase-drawer/PremiumTrackPurchaseDrawer.tsx:156]: Planning to hook this up later? There were already existing places on web that linked out to terms of use which I was able to reuse.

File: packages/mobile/src/components/premium-track-purchase-drawer/StripePurchaseConfirmationButton.tsx [hunk 4]
```diff
@@ -1,74 +1,64 @@
...
 export const StripePurchaseConfirmationButton = ({
+  trackId,
   price
 }: StripePurchaseConfirmationButtonProps) => {
   const dispatch = useDispatch()
-  const navigation = useNavigation()
   const { specialLightGreen1 } = useThemeColors()
-  const ethWallet = useSelector(getAccountERCWallet)
+  const stage = useSelector(getPurchaseContentFlowStage)
+  const isLoading =
```
Comment (schottra) [inline packages/mobile/src/components/premium-track-purchase-drawer/StripePurchaseConfirmationButton.tsx:37]: There's a [util in common ](https://github.com/AudiusProject/audius-client/blob/41cc4af1a1d329186afe8ff2b8452571b1745cf8/packages/common/src/store/purchase-content/utils.ts#L3)now that you can use for this.

**Review by schottra — APPROVED** [review]
Looks good!

**Comment by audius-infra** [comment]
It looks like there may be some changes to native mobile code, which requires triggering a full app release. Please follow the instructions here: https://www.notion.so/audiusproject/When-to-bump-app-version-2644a8f772364a4d91f44abcba44ce0b?pvs=4. cc @nicoback2 @sliptype

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/rt-wire-up-usdc-mobile

File: packages/common/src/store/ui/stripe-modal/sagas.ts (hunk 2 above)
Comment (rickyrombo) [reply packages/common/src/store/ui/stripe-modal/sagas.ts:43]: It's not related to my change. Ideally, we wouldn't set this manually like this but have some hook generate the `'closing'` vs `false` for you, and you don't have to see or understand the difference. `'closing'` is distinct from `false` in that `'closing'` means that it's in the _process_ of closing (animating closed) and `false` is that it's fully disappeared. This allowed @dylanjeffers to do [conditional rendering] of Modals to improve perf.

**Pull request merged**